File: microservices/investment/src/main/java/com/illunex/invest/investment/persistence/entity/EvaluateReviewItemCategory.java
package com.illunex.invest.investment.persistence.entity;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "evaluate_review_item_category")
public class EvaluateReviewItemCategory {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long idx;
    String category;
    Integer weight;

    @ManyToOne
    @JoinColumn(name = "template_idx")
    EvaluateReviewItemTemplate reviewItemTemplate;

    @OneToMany(mappedBy = "reviewItemCategory", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    List<EvaluateReviewItem> reviewItem;
}

File: microservices/invest/src/main/kotlin/com/illunex/invest/invest/persistence/entity/EvaluateReviewItemCategory.kt
package com.illunex.invest.invest.persistence.entity

import javax.persistence.*

@Entity
@Table(name = "evaluate_review_category")
data class EvaluateReviewItemCategory(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var idx: Long? = null,
    var vcCompanyIdx: Long? = null,
    var category: String? = null,

    @OneToOne(mappedBy = "category", fetch = FetchType.LAZY, cascade = [CascadeType.DETACH])
    var evaluate: Evaluate?,

    @OneToMany(mappedBy = "category", cascade = [CascadeType.ALL])
    var reviewItem: List<EvaluateReviewItem>?
) {
    constructor() : this(null, null, null, null, null )
}
